setup.py

```diff
@@ -74,7 +74,7 @@ def versionfromfile(*filepath):
     license="Apache",
     url="https://github.com/deepset-ai/haystack",
     download_url=f"https://github.com/deepset-ai/haystack/archive/{_version}.tar.gz",
-    packages=find_packages(exclude=["*.tests", "*.tests.*", "tests.*", "tests"]),
+    packages=find_packages(exclude=["*.rest_api", "*.rest_api.*", "rest_api.*", "rest_api"]),
```

Comment:
Also this line leaves me quite confused :sweat_smile: Isn't enough to specify the folder to ignore in one of these four ways (probably the last)?

I also think it might be better to use a whitelist and only add `haystack` to the list of folders to include. That might make more sense.